pkgs/tools/networking/libqofono/default.nix

```
@@ -0,0 +1,58 @@
+{ lib
+, substituteAll
+, mkDerivation
+, fetchFromGitLab
+, mobile-broadband-provider-info
+, qmake
+, qtbase
+, qtdeclarative
+}:
+
+mkDerivation rec {
+  pname = "libqofono";
+  version = "0.103";
+
+  src = fetchFromGitLab {
+    domain = "git.sailfishos.org";
+    owner = "mer-core";
+    repo = "libqofono";
+    rev = version;
+    sha256 = "1ly5aj412ljcjvhqyry6nhiglbzzhczsy1a6w4i4fja60b2m1z45";
+  };
+
+  patches = [
+    (substituteAll {
+      src = ./0001-NixOS-provide-mobile-broadband-provider-info-path.patch;
+      mobile_broadband_provider_info = mobile-broadband-provider-info;
```

I would prefer changing the name in the patches to use
```suggestion
      inherit mobile-broadband-provider-info;
```
but that's up to you.